File: dwm/naive.py
class row:
	
	def __init__(self):
		self.chills=""
		self.runny_nose=""
		self.headache=""
		self.flue=""



def calculate(clas,row_array,unknown,n):

	c_chills=0
	c_runny_nose=0
	c_headache=0
	c_class=0

	for i in range(n):


		if row_array[i].flue==clas:

			c_class = c_class + 1
			if row_array[i].chills == unknown.chills:
				c_chills=c_chills+1
			if row_array[i].runny_nose == unknown.runny_nose:
				c_runny_nose=c_runny_nose+1
			if row_array[i].headache == unknown.headache:
				c_headache=c_headache+1


	ans = (float(c_chills*c_runny_nose*c_headache)/(c_class**3))*(float(c_class)/n)
	return ans



n = int(input("enter the number of tuples"))

row_array=[]
print("enter the tuples")

for i in range(n):
	print("enter the input value for tuple",i)
	
	r = row()
	r.chills = input()
	r.runny_nose = input()
	r.headache = input()
	r.flue = input() 
	row_array.append(r)
r=1
while r==1:

	print("enter the unknown values")
	unknown=row()
	unknown.chills = input()
	unknown.runny_nose = input()
	unknown.headache = input()
	
	y=calculate('1',row_array,unknown,n)
	n=calculate('0',row_array,unknown,n)
	print("The probability of yes is ",y)
	print("The probability of no is ",n)

	print("do you wanna continue")
	r=int(input())

File: dwm/naive.c
#include<stdio.h>
#include<stdlib.h>

struct row
{
	char chills;
	char runny_nose;
	char headache;
	char flu;
}data[20];



float calculate(char,struct row[],int,struct row);

void main()
{
	int n,i,r=1;
	char ch;
	printf("\nenter the number of tuples");
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		printf("\nenter the chills,runny_nose,headache,flu");
		scanf("%c%c%c%c",&data[i].chills,&data[i].runny_nose,&data[i].headache,&data[i].flu);		
	}
	while(r==1)
	{
		struct row unknown;
		printf("Enter the unknown sample");
		scanf("%c%c%c",&unknown.chills,&unknown.runny_nose,&unknown.headache);
		float y = calculate('1',data,n,unknown);
		float n = calculate('0',data,n,unknown);
		printf("y= %f \n n= %f \n",y,n);
		printf("\nDo you want to continue");
		scanf("%d",&r);
	}
}


float calculate(char class,struct row data[],int n,struct row unknown)
{
	int i;
	int c_1=0,c_2=0,c_3=0,class_counter=0;
	for(i=0;i<n;i++)
	{

		if(data[i].flu==class)
		{
			class_counter++;
			if(unknown.chills==data[i].chills)
			{
				c_1++;
			}
			if(unknown.runny_nose==data[i].runny_nose)
			{
				c_2++;
			}
			if(unknown.headache==data[i].headache)
			{
				c_3++;
			}
		}
		float ans = ((float)(c_1*c_2*c_3)/(class_counter*class_counter*class_counter*class_counter))*((float)class_counter/n);
		return ans;
	}
}